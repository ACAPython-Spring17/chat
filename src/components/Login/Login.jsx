import React, { Component } from 'react';
import PropTypes from 'prop-types';

import TextField from 'material-ui/TextField';

class Login extends Component {
  state = {
    error: false,
  };

  handleKeyPress = (event) => {
    const { onLogin } = this.props;
    const value = event.target.value.trim();

    if (event.key === 'Enter') {
      if (value) {
        onLogin(value);
        this.setState({ error: false });
      } else {
        this.setState({ error: true });
      }
    }
  };

  render() {
    const { error } = this.state;

    return (
      <div className="grid-x align-center">
        <TextField
          name="username"
          floatingLabelText="Username"
          hintText="Enter your username"
          errorText={error ? 'Username cannot be empty' : ''}
          onKeyPress={this.handleKeyPress}
        />
      </div>
    );
  }
}

Login.propTypes = {
  onLogin: PropTypes.func.isRequired,
}

export default Login;
