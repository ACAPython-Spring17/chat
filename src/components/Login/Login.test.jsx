import React from 'react';
import TextField from 'material-ui/TextField';
import Login from './Login';

describe('Login', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Login />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('when enter key is pressed', () => {
    describe('when value is empty', () => {
      it('should render en error', () => {
        const wrapper = shallow(<Login />);
        const event = {
          target: {
            value: '  ',
          },
          key: 'Enter',
        };

        wrapper.find(TextField).simulate('keyPress', event);

        expect(wrapper.find(TextField).prop('errorText'))
          .toEqual('Username cannot be empty');
      });
    });

    describe('when value is not empty', () => {
      it('should call props.onLogin and remove error', () => {
        const onLogin = jest.fn();

        const wrapper = shallow(<Login onLogin={onLogin} />);
        const event = {
          target: {
            value: 'some value',
          },
          key: 'Enter',
        };

        wrapper.find(TextField).simulate('keyPress', event);

        expect(wrapper.find(TextField).prop('errorText')).toEqual('');
        expect(onLogin).toHaveBeenCalledWith('some value');
      });
    });
  });
});
