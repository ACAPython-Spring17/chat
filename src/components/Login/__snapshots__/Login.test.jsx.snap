// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render correctly 1`] = `
<div
  className="grid-x align-center"
>
  <TextField
    disabled={false}
    errorText=""
    floatingLabelFixed={false}
    floatingLabelText="Username"
    fullWidth={false}
    hintText="Enter your username"
    multiLine={false}
    name="username"
    onKeyPress={[Function]}
    rows={1}
    type="text"
    underlineShow={true}
  />
</div>
`;
